utils/trace/trace_test.go

```diff
@@ -0,0 +1,74 @@
+package trace
+
+import (
+	"bytes"
+	"log"
+	"testing"
+
+	. "gopkg.in/check.v1"
+)
+
+type TraceSuite struct {
+	buf bytes.Buffer
+}
+
+var _ = Suite(&TraceSuite{})
+
+func Test(t *testing.T) { TestingT(t) }
+
+func (s *TraceSuite) SetUpSuite(c *C) {
+	SetLogger(log.New(&s.buf, "", 0))
+}
+
+func (s *TraceSuite) TearDownSuite(c *C) {
+	SetLogger(newLogger())
+}
+
+func (s *TraceSuite) TearDownTest(c *C) {
+	SetTarget(0)
+	s.buf.Reset()
+}
+
+func (s *TraceSuite) TestEmpty(c *C) {
+	General.Print("test")
+	c.Assert(s.buf.String(), Equals, "")
+}
+
+func (s *TraceSuite) TestOneTarget(c *C) {
+	SetTarget(General)
+	General.Print("test")
+	c.Assert(s.buf.String(), Equals, "test\n")
+}
+
+func (s *TraceSuite) TestMultipleTargets(c *C) {
+	SetTarget(General | Packet)
+	General.Print("a")
+	Packet.Print("b")
+	c.Assert(s.buf.String(), Equals, "a\nb\n")
+}
+
+func (s *TraceSuite) TestPrintf(c *C) {
+	SetTarget(General)
+	General.Printf("a %d", 1)
+	c.Assert(s.buf.String(), Equals, "a 1\n")
+}
+
+func (s *TraceSuite) TestDisabledMultipleTargets(c *C) {
+	SetTarget(General)
+	General.Print("a")
+	Packet.Print("b")
+	c.Assert(s.buf.String(), Equals, "a\n")
+}
+
+func (s *TraceSuite) BenchmarkDisabledTarget(c *C) {
+	for i := 0; i < c.N; i++ {
+		General.Print("test")
+	}
+}
+
+func (s *TraceSuite) BenchmarkEnabledTarget(c *C) {
+	SetTarget(General)
```

Comment:
For the purposes of the benchmark, I think the logger could write to `io.Discard` as we only want to understand the overall overhead.

```suggestion
	SetLogger(log.New(io.Discard, "", 0))
	SetTarget(General)
```

This does not seem to affect `check.v1` (as it probably wraps/ignore stderr), with standard benchmark, the call costs seems to be:
~1900 ns/op (default stderr)
~40 ns/op (io.Discard)
~300 ns/op (file on disk)